Build failing on format-truncation in doomclassic

While compiling from source on Arch Linux (using `master`: https://github.com/RobertBeckebans/RBDOOM-3-BFG/commit/4d099cd62bd537c69c49448737769ad34ba6a4dc), I ran into the following issue:

```c
[  0%] Creating idlib/precompiled.h.gch for idlib
[  0%] Built target precomp_header_idlib
[  9%] Built target idlib
[  9%] Creating idlib/precompiled.h.gch for RBDoom3BFG
[  9%] Built target precomp_header_rbdoom3bfg
Consolidate compiler generated dependencies of target RBDoom3BFG
[  9%] Building CXX object CMakeFiles/RBDoom3BFG.dir/home/Code/RBDOOM-3-BFG/doomclassic/doom/hu_stuff.cpp.o
/home/Code/RBDOOM-3-BFG/doomclassic/doom/hu_stuff.cpp: In function ‘void HU_Init()’:
/home/Code/RBDOOM-3-BFG/doomclassic/doom/hu_stuff.cpp:330:60: error: ‘snprintf’ output truncated before the last format character [-Werror=format-truncation=]
  330 |                 snprintf(buffer, strlen(buffer), "STCFN%.3d", j++);
      |                                                            ^
/home/Code/RBDOOM-3-BFG/doomclassic/doom/hu_stuff.cpp:330:25: note: ‘snprintf’ output 9 bytes into a destination of size 8
  330 |                 snprintf(buffer, strlen(buffer), "STCFN%.3d", j++);
      |                 ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: some warnings being treated as errors
make[2]: *** [CMakeFiles/RBDoom3BFG.dir/build.make:6054: CMakeFiles/RBDoom3BFG.dir/home/Code/RBDOOM-3-BFG/doomclassic/doom/hu_stuff.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:159: CMakeFiles/RBDoom3BFG.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

`g++ (GCC) 12.2.0`

--- a/doomclassic/doom/f_finale.cpp
+++ b/doomclassic/doom/f_finale.cpp
@@ -725,7 +725,8 @@ void F_BunnyScroll (void)
     int		x;
     patch_t*	p1;
     patch_t*	p2;
-    char	name[10];
+    size_t	name_len = 10;
+    char	name[name_len];
     int		stage;
 		
     p1 = (patch_t*)W_CacheLumpName ("PFUB2", PU_LEVEL_SHARED);
@@ -765,8 +766,8 @@ void F_BunnyScroll (void)
 	S_StartSound (NULL, sfx_pistol);
 	::g->laststage = stage;
     }
-	
-    snprintf (name, strlen(name), "END%i",stage);
+
+	snprintf(name, name_len, "END%i", stage);
     V_DrawPatch ((ORIGINAL_WIDTH-13*8)/2, (ORIGINAL_HEIGHT-8*8)/2,0, (patch_t*)W_CacheLumpName (name,PU_CACHE_SHARED));
 }
 

--- a/doomclassic/doom/hu_stuff.cpp
+++ b/doomclassic/doom/hu_stuff.cpp
@@ -319,7 +319,8 @@ void HU_Init(void)
 
 	int		i;
 	int		j;
-	char	buffer[9];
+	size_t	buffer_len = 9;
+	char	buffer[buffer_len];
 
 	shiftxform = english_shiftxform;
 
@@ -327,7 +328,7 @@ void HU_Init(void)
 	j = HU_FONTSTART;
 	for (i=0;i<HU_FONTSIZE;i++)
 	{
-		snprintf(buffer, strlen(buffer), "STCFN%.3d", j++);
+		snprintf(buffer, buffer_len, "STCFN%03d", j++);
 		::g->hu_font[i] = (patch_t *) W_CacheLumpName(buffer, PU_STATIC_SHARED);
 	}
 

